restapi/src/main/java/io/stargate/web/resources/v2/schemas/UserDefinedTypesResource.java

```diff
@@ -0,0 +1,388 @@
+      })
+  public Response createType(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam(HEADER_TOKEN_AUTHENTICATION)
+          final String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(value = "", required = true) @NotNull final UserDefinedTypeAdd udtAdd,
+      @Context final HttpServletRequest request) {
+
+    return RequestHandler.handle(
+        () -> {
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, getAllHeaders(request));
+          Keyspace keyspace = authenticatedDB.getDataStore().schema().keyspace(keyspaceName);
+
+          if (keyspace == null) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        "keyspace does not exists", Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+          String typeName = udtAdd.getName();
+          if (Strings.isNullOrEmpty(typeName)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        "Type name must be provided", Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+
+          db.getAuthorizationService()
+              .authorizeSchemaWrite(
+                  authenticatedDB.getAuthenticationSubject(),
+                  keyspaceName,
+                  null,
+                  Scope.CREATE,
+                  SourceAPI.REST,
+                  ResourceKind.TYPE);
+
+          List<Column> columns = new ArrayList<>();
+          for (UserDefinedTypeField colDef : udtAdd.getFieldDefinitions()) {
+            String fieldName = colDef.getName();
+            String typeDef = colDef.getTypeDefinition();
+            if (Strings.isNullOrEmpty(fieldName) || Strings.isNullOrEmpty(typeDef)) {
+              return Response.status(Response.Status.BAD_REQUEST)
+                  .entity(
+                      new Error(
+                          "Type name and definition must be provided.",
+                          Response.Status.BAD_REQUEST.getStatusCode()))
+                  .build();
+            }
+            columns.add(
+                Column.create(
+                    fieldName,
+                    Kind.Regular,
+                    Type.fromCqlDefinitionOf(keyspace, colDef.getTypeDefinition())));
+          }
+
+          UserDefinedType udt =
+              ImmutableUserDefinedType.builder()
+                  .keyspace(keyspaceName)
+                  .name(typeName)
+                  .addColumns(columns.toArray(new Column[0]))
```

Comment:
Should we validate that there's at least one column? If there were no fieldDefinitions then columns could be empty here, right?

Reply:
Yes, I added a check and test.